src/compas_timber/connections/t_step_joint.py

```diff
@@ -132,22 +119,22 @@ def main_extension_plane(self):
         ref_side_index = min(ref_side_dict, key=ref_side_dict.get)
         return self.cross_beam.opp_side(ref_side_index)
 
-    def set_step_depths(self, step_depth=None, heel_depth=None):
+    def set_step_depths(self):
         """Sets the default step and heel depths based on the joint type if they are not provided."""
+        self.step_shape = self.step_shape or 0 # Set value for step_shape
+
         if self.step_shape == 0:  # 'step' shape
-            step_depth = step_depth if step_depth is not None else self.cross_beam.height / 4
-            heel_depth = 0.0
+            self.step_depth = self.step_depth or self.cross_beam.height / 4
```

Comment:
since I imagine that `self.step_depth` can be `0` I'd say we should indeed check for `is None` here.